amacc.c

```diff
@@ -442,41 +443,33 @@ void expr(int lev)
         while (tk == Mul) { next(); ty += PTR; }
         if (tk == ')') next();
         else fatal("close paren expected in sizeof");
-        *++e = IMM; *++e = ty >= PTR ? sizeof(int) : tsize[ty];
+        *--n = ty >= PTR ? sizeof(int) : tsize[ty]; *--n = Num;
         ty = INT;
         break;
     case Id:
         d = id; next();
         // function call
         if (tk == '(') {
+            if (d->class != Syscall && d->class != Func) { fatal("bad function call");}
```

Comment:
[style] Should be `if (d->class != Syscall && d->class != Func) fatal("bad function call");`